<launch>


	 <!-- declare arguments with default values -->
  <arg name="pcap" default="" />
  <arg name="calibration" default="$(find velodyne_pointcloud)/params/32db.yaml"/>

  <!-- start nodelet manager and driver nodelets -->
  <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
    <arg name="model" value="32E"/>
    <arg name="pcap" value="$(arg pcap)"/>
  </include>

  <!-- start cloud nodelet -->
  <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
    <arg name="calibration" value="$(arg calibration)"/>
	<arg name="min_range" value="0.5" />
	<arg name="max_range" value="100.0" />
  </include>

	<node pkg="velodyne_node" name="velodyne_node" type="velodyne_node" output="screen" >
			<param name="ring" value="15" />
			<param name="pointcloud_in" value="/velodyne_points" />
			<param name="pointcloud_out" value="/pointcloud_out" />
	</node>


  
	<node pkg="hough_2lines_detect" name="hough_2lines_detect" type="hough_2lines_detect" output="screen" >
			<param name="in" value="/pointcloud_out" />
			<param name="line_out" value="/2lines"/>
			<param name="line_out1" value="/1lines"/>
	</node>

  
	<node pkg="draw_marker_lines" name="draw_marker_lines_right" type="draw_marker_lines_node" output="screen" >
			<param name="line_topic" value="/2lines" />
			<param name="draw_topic" value="/draw_2lines" />
	</node>


	<node pkg="draw_marker_lines" name="draw_marker_lines_left" type="draw_marker_lines_node" output="screen" >
			<param name="line_topic" value="/1lines" />
			<param name="draw_topic" value="/draw_1_lines" />
	</node>

</launch>

